src/libpriv/rpmostree-util.c

```diff
@@ -587,3 +588,65 @@ rpmostree_file_get_path_cached (GFile *file)
 
   return path;
 }
+
+gboolean
+rpmostree_deployment_get_layered_info (OstreeRepo        *repo,
+                                       OstreeDeployment  *deployment,
+                                       gboolean          *out_is_layered,
+                                       char             **out_base_layer,
+                                       char            ***out_layered_pkgs,
+                                       GError           **error)
+{
+  g_autoptr(GVariant) commit = NULL;
+  g_autoptr(GVariant) metadata = NULL;
+  g_autoptr(GVariantDict) dict = NULL;
+  g_auto(GStrv) layered_pkgs = NULL;
+  const char *csum = ostree_deployment_get_csum (deployment);
+  gboolean is_layered = FALSE;
+  g_autofree char *base_layer = NULL;
+
+  if (!ostree_repo_load_commit (repo, csum, &commit, NULL, error))
+    return FALSE;
+
+  metadata = g_variant_get_child_value (commit, 0);
+  dict = g_variant_dict_new (metadata);
+
+  /* More recent versions have an explicit clientlayer attribute (which
+   * realistically will always be TRUE). For older versions, we just
+   * rely on the treespec being present. */
+  if (!g_variant_dict_lookup (dict, "rpmostree.clientlayer", "b", &is_layered))
+    is_layered = g_variant_dict_contains (dict, "rpmostree.spec");
+
+  /* only fetch base if we have to */
+  if (is_layered && out_base_layer != NULL)
+    {
+      base_layer = ostree_commit_get_parent (commit);
+      g_assert (base_layer);
+    }
+
+  /* only fetch the pkgs if we have to */
+  if (is_layered && out_layered_pkgs != NULL)
+    {
+      g_autoptr(GVariant) treespec_v = NULL;
+      g_autoptr(GVariantDict) treespec = NULL;
+
+       /* there should always be a treespec */
+      treespec_v = g_variant_dict_lookup_value (dict, "rpmostree.spec",
```

Comment:
I think this is my main concern - basically the "treespec" stuff feels hacky to me.  This will kind of lock us in to supporting it forever.  Or at least the subset that we read back out here.

That said...I think let's go for it.  We just have to keep this in mind.